stdlib/sql/snowflake.go

```diff
@@ -0,0 +1,201 @@
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewInt(d)
+			case flux.TBool:
+				b, err := strconv.ParseBool(value)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewBool(b)
+			default:
+				row[i] = values.New(value)
+			}
+		case time.Time:
+			// DATE, TIME and TIMESTAMP_NTZ types get scanned to time.Time by the driver,
+			// but they have no counterpart in Flux therefore will be represented as string
+			switch m.sqlTypes[i].DatabaseTypeName() {
+			case "DATE":
+				row[i] = values.NewString(value.Format(layoutDate))
+			case "TIME":
+				row[i] = values.NewString(value.Format(layoutTime))
+			case "TIMESTAMP_NTZ":
+				row[i] = values.NewString(value.Format(layoutTimeStampNtz))
+			default:
+				row[i] = values.NewTime(values.ConvertTime(value))
+			}
+		case nil:
+			row[i] = values.NewNull(flux.SemanticType(m.columnTypes[i]))
+		default:
+			execute.PanicUnknownType(flux.TInvalid)
+		}
+	}
+	return row, nil
+}
+
+func (m *SnowflakeRowReader) InitColumnNames(names []string) {
+	m.columnNames = names
+}
+
+func (m *SnowflakeRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	fluxTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "FIXED", "NUMBER": // FIXED is reported by Snowflake driver
+			_, scale, ok := types[i].DecimalSize()
+			if ok && scale > 0 {
+				fluxTypes[i] = flux.TFloat
+			} else {
+				fluxTypes[i] = flux.TInt
+			}
+		case "REAL", "FLOAT": // REAL is reported by Snowflake driver
+			fluxTypes[i] = flux.TFloat
+		case "BOOLEAN":
+			fluxTypes[i] = flux.TBool
+		case "TIMESTAMP_TZ", "TIMESTAMP_LTZ": // "TIMESTAMP_NTZ", "DATE" and "TIME" will be represented as string
+			fluxTypes[i] = flux.TTime
+		default:
+			fluxTypes[i] = flux.TString
+		}
+	}
+	m.columnTypes = fluxTypes
+	m.sqlTypes = types
+}
+
+func (m *SnowflakeRowReader) ColumnNames() []string {
+	return m.columnNames
+}
+
+func (m *SnowflakeRowReader) ColumnTypes() []flux.ColType {
+	return m.columnTypes
+}
+
+func (m *SnowflakeRowReader) SetColumnTypes(types []flux.ColType) {
+	m.columnTypes = types
+}
+
+func (m *SnowflakeRowReader) SetColumns(i []interface{}) {
+	m.columns = i
+}
+
+func (m *SnowflakeRowReader) Close() error {
+	if m.CloseFunc != nil {
+		return m.CloseFunc()
+	}
+	if err := m.Cursor.Err(); err != nil {
+		return err
+	}
+	return m.Cursor.Close()
+}
+
+func NewSnowflakeRowReader(r *sql.Rows) (execute.RowReader, error) {
+	reader := &SnowflakeRowReader{
+		Cursor: r,
+	}
+	cols, err := r.Columns()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnNames(cols)
+
+	types, err := r.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnTypes(types)
+
+	return reader, nil
+}
+
+// SnowflakeTranslateColumn translates flux colTypes into their corresponding Snowflake column type
+func SnowflakeColumnTranslateFunc() translationFunc {
+	c := map[string]string{
```

Comment:
Could you please hoist this value definition up and out of the function itself? It's effectively static, so we should define it up above.